### PR TITLE
feat: Telemetry for Events (`SDK` metadata)

### PR DESCRIPTION
**Pull request by fabriziodemaria**
- Sending events also attaches metadata about the SDK ID (`SDK_ID_JAVA_CONFIDENCE`) and SDK VERSION
- Resolving flags emits unchanged metadata about SDK ID (`SDK_ID_JAVA_PROVIDER`) and SDK VERSION 

The SDK ID is used to identify the main logic adopted to perform flag resolving and event sending, rather than reflecting 1:1 the artifact structuring. If/When we will open the flag-resolve API that don't use OpenFeature, the metadata should be changed accordingly to `SDK_ID_JAVA_CONFIDENCE` allowing us to visualize adoption o the different user-facing APIs

Note: invalidates https://github.com/spotify/confidence-openfeature-provider-java/pull/105

### CODE REVIEW

File: src/main/java/com/spotify/confidence/GrpcEventUploader.java [hunk 1]
```diff
@@ -23,6 +28,17 @@ class GrpcEventUploader implements EventUploader {
     this.managedChannel = managedChannel;
     this.stub = EventsServiceGrpc.newFutureStub(managedChannel);
     this.clock = clock;
+    try {
+      final Properties prop = new Properties();
+      prop.load(this.getClass().getResourceAsStream("/version.properties"));
```
Comment (vahidlazio) [inline src/main/java/com/spotify/confidence/GrpcEventUploader.java:33]: we are using the same block to get the version of the sdk in [here](https://github.com/spotify/confidence-openfeature-provider-java/blob/8271811da17a59d30539900102ba78bf9f1bff2d/src/main/java/com/spotify/confidence/GrpcFlagResolver.java#L29), do you think if it makes sense to extract this part to a utility function giving back the sdk version?

File: src/main/proto/confidence/flags/resolver/v1/types.proto [hunk 2]
```diff
@@ -60,4 +60,20 @@ enum SdkId {
   SDK_ID_PYTHON_PROVIDER = 6;
   // Confidence OpenFeature GO Provider.
   SDK_ID_GO_PROVIDER = 7;
+  // Confidence OpenFeature Ruby Provider.
```
Comment (vahidlazio) [inline src/main/proto/confidence/flags/resolver/v1/types.proto:63]: curious why are we adding these types back?

Comment (vahidlazio) [reply src/main/proto/confidence/flags/resolver/v1/types.proto:63]: no all good 👍

Comment (fabriziodemaria) [reply src/main/proto/confidence/flags/resolver/v1/types.proto:63]: I tend to agree with you both that the pros of keeping the protos perfectly in sync are not as relevant as the cons of adding unused generated code to the client, as well as giving the client definitions that it is perhaps not meant to be using... I'd keep this small PR as it is, but it is an interesting topic though

File: src/main/java/com/spotify/confidence/GrpcFlagResolver.java [hunk 3]
```diff
@@ -4,35 +4,27 @@
...
-      final Properties prop = new Properties();
-      prop.load(this.getClass().getResourceAsStream("/version.properties"));
-      this.SDK_VERSION = prop.getProperty("version");
-    } catch (IOException e) {
-      throw new RuntimeException("Can't determine version of the SDK", e);
-    }
-
+    this.sdk =
+        Sdk.newBuilder()
+            .setId(SdkId.SDK_ID_JAVA_CONFIDENCE)
```
Comment (nicklasl) [inline src/main/java/com/spotify/confidence/GrpcFlagResolver.java:25]: This should be `SDK_ID_JAVA_PROVIDER` right?
That's what the PR description states anyway. :)

Comment (fabriziodemaria) [reply src/main/java/com/spotify/confidence/GrpcFlagResolver.java:25]: Correct, I must have forgotten to push the change

**Review by nicklasl — APPROVED** [review]

**Review by vahidlazio — APPROVED** [review]

**Pull request merged**